src/OpenLoco/Entities/EntityTweener.cpp

```diff
@@ -0,0 +1,114 @@
+#include "EntityTweener.h"
+#include <cmath>
+#include <iostream>
+
+namespace OpenLoco
+{
+    using EntityListType = EntityManager::EntityListType;
+    using EntityListIterator = EntityManager::ListIterator<EntityBase, &EntityBase::next_thing_id>;
+
+    template<EntityListType id>
+    void PopulateEntities(std::vector<EntityBase*>& list, std::vector<Map::map_pos3>& posList)
+    {
+        auto entsView = EntityManager::EntityList<EntityListIterator, id>();
+        for (auto* ent : entsView)
+        {
+            list.push_back(ent);
+            posList.emplace_back(ent->x, ent->y, ent->z);
+        }
+    }
+
+    static EntityTweener _tweener;
+
+    EntityTweener& EntityTweener::Get()
+    {
+        return _tweener;
+    }
+
+    void EntityTweener::PreTick()
+    {
+        Restore();
+        Reset();
+        PopulateEntities<EntityListType::misc>(_entities, _prePos);
+        PopulateEntities<EntityListType::vehicleHead>(_entities, _prePos);
+        PopulateEntities<EntityListType::vehicle>(_entities, _prePos);
```

Comment:
I'm still not sure this is safe to do. This is going to tween veh1, veh2, tail as well. For one its a waste of resource invalidating them as they don't physically exist and for two their positions are not guaranteed to be related to drawing. It would mean though that you can't use the same interface for misc and for vehicles though.